public/pages/workflows/workflow_list/resource_list.tsx

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useEffect, useCallback } from 'react';
+import { columns } from '../../workflow_detail/tools/resources/columns';
+import {
+  EuiCodeBlock,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiBasicTable,
+  EuiButtonIcon,
+  RIGHT_ALIGNMENT,
+  EuiText,
+  Direction,
+} from '@elastic/eui';
+import {
+  Workflow,
+  WorkflowResource,
+  customStringify,
+} from '../../../../common';
+import {
+  getIndex,
+  getIngestPipeline,
+  getSearchPipeline,
+  useAppDispatch,
+} from '../../../store';
+import { getDataSourceId } from '../../../utils';
+
+interface ResourceListProps {
+  workflow?: Workflow;
+}
+
+/**
+ * The searchable list of resources for a particular workflow.
+ */
+export function ResourceList(props: ResourceListProps) {
+  const [allResources, setAllResources] = useState<WorkflowResource[]>([]);
+  const dispatch = useAppDispatch();
+  const dataSourceId = getDataSourceId();
+  const [itemIdToExpandedRowMap, setItemIdToExpandedRowMap] = useState<{
+    [key: string]: React.ReactNode;
+  }>({});
+  const [codeBlockData, setCodeBlockData] = useState<any>(null);
+  const [pageIndex, setPageIndex] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
+  const [sortField, setSortField] = useState<keyof WorkflowResource>('id');
+  const [sortDirection, setSortDirection] = useState<Direction>('asc');
+
+  // Hook to initialize all resources. Reduce to unique IDs, since
+  // the backend resources may include the same resource multiple times
+  // (e.g., register and deploy steps persist the same model ID resource)
+  useEffect(() => {
+    if (props.workflow?.resourcesCreated) {
+      const resourcesMap: { [id: string]: WorkflowResource } = {};
+      props.workflow.resourcesCreated.forEach((resource) => {
+        resourcesMap[resource.id] = resource;
+      });
+      setAllResources(Object.values(resourcesMap));
+    }
+  }, [props.workflow?.resourcesCreated]);
+
+  useEffect(() => {
+    if (codeBlockData) {
+      const { item, data } = codeBlockData;
+      setItemIdToExpandedRowMap((prevMap) => ({
+        ...prevMap,
+        [item.id]: renderExpandedRow(data),
+      }));
+    }
+  }, [codeBlockData]);
```

Comment:
Few enhancements I can see here:
1. I don't think we need this extra custom hook. It seems we can update the `itemIdToExpandedRowMap` with the data directly on line 109, and skip using the additional `codeBlockData` state var entirely
2. We shouldn't need a `codeBlockData` state at all. Instead, we should just dynamically render the code block based on the item ID exposed in the `render` fn for the table column directly on line 160.

Reply:
Removed.